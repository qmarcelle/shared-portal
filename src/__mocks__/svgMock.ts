import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
