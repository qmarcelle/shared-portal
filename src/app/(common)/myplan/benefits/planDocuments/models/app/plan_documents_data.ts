import { VisibilityRules } from '@/visibilityEngine/rules';

export interface PlanDocumentsData {
  visibilityRules?: VisibilityRules;
  iframeContent?: string;
}
