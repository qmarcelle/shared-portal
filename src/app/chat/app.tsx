'use client';

import { ChatWidget } from './components/core/ChatWidget';
import { ChatContextProvider } from './components/providers/ChatContextProvider';
import type { ChatConfig, GenesysUserData } from './types/types';

// Default config for development - should be overridden in production
const defaultConfig: ChatConfig = {
  token: process.env.NEXT_PUBLIC_CHAT_TOKEN || '',
  endPoint: process.env.NEXT_PUBLIC_CHAT_ENDPOINT || '',
  opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
  userID: '',
  memberFirstname: '',
  memberLastname: '',
  memberId: '',
  groupId: '',
  planId: '',
  planName: '',
  businessHours: {
    isOpen24x7: false,
    days: [],
    timezone: 'America/New_York',
    isCurrentlyOpen: true,
    lastUpdated: Date.now(),
    source: 'default',
  },
};

// Default user data for development - should be overridden in production
const defaultUserData: Partial<GenesysUserData> = {
  config: {
    deploymentId: process.env.NEXT_PUBLIC_GENESYS_DEPLOYMENT_ID || '',
    region: process.env.NEXT_PUBLIC_GENESYS_REGION || '',
  },
  userInfo: {
    firstName: '',
    lastName: '',
  },
};

export default function ChatPage() {
  return (
    <ChatContextProvider config={defaultConfig} userData={defaultUserData}>
      <div className="chat-container">
        <ChatWidget />
      </div>
    </ChatContextProvider>
  );
}
