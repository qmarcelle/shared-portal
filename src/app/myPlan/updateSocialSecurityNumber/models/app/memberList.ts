export type MemberList = {
  members: Member[];
};

export type Member = {
  memberCk: number;
  firstName: string;
  lastName: string;
  birthDate: string;
  hasSocial: boolean;
};
