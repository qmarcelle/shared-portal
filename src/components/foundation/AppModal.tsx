import FocusTrap from 'focus-trap-react';
import Image from 'next/image';
import React, { ReactElement, createContext } from 'react';
import { Modal } from 'react-responsive-modal';
import { create } from 'zustand';
import leftIcon from '../../../public/assets/left.svg';
import { IComponent } from '../IComponent';
import { Column } from './Column';
import { bcbstSilhouletteLogo, closeIcon } from './Icons';
import { Row } from './Row';
import { Spacer } from './Spacer';
import { TextBox } from './TextBox';

/* eslint-disable @typescript-eslint/no-explicit-any */
interface ModalProps extends IComponent {
  content: ReactElement & ModalChildProps;
  store?: any;
  isChildActionAppModal?: boolean;
  childModalContent?: ReactElement & ModalChildProps;
}
interface ModalControllerProps {
  showBack: boolean;
  showModal: boolean;
  pageIndex: number;
  store: null;
  isChildAction?: boolean;
  updateShowBack: (isShowBack: boolean) => void;
  updateshowModal: (isShowBack: boolean) => void;
  updatepageIndex: (pageIndex: number) => void;
  updateStore: (store: null) => void;
  showAppModal: (props: ModalProps) => void;
  dismissModal: () => void;
  isChildModal: boolean;
  showChildModal: () => void;
  dismissChildModal: () => void;
  isChildActionAppModalProp: boolean;
}

export const useAppModalStore = create<ModalControllerProps>((set) => ({
  showBack: false,
  showModal: false,
  isChildModal: false,
  pageIndex: 0,
  store: null,
  isChildActionAppModalProp: false,
  updateShowBack: (isShowBack: boolean) =>
    set(() => ({ showBack: isShowBack })),
  updateshowModal: (isShowModal: boolean) =>
    set(() => ({ showModal: isShowModal })),
  updatepageIndex: (pageIndex: number) => set(() => ({ pageIndex: pageIndex })),
  updateStore: (store: null) => set(() => ({ store: store })),
  showAppModal: ({
    content,
    store,
    isChildActionAppModal,
    childModalContent,
  }: ModalProps) => {
    modalBody = content;
    childModalBody = childModalContent;
    if (isChildActionAppModal) {
      set(() => ({
        isChildActionAppModalProp: isChildActionAppModal,
      }));
    }
    set(() => ({
      showModal: true,
    }));
    set(() => ({
      store: store,
    }));
    pageIndexStack = [0];
  },
  dismissModal: () => {
    set(() => ({
      showModal: false,
    }));
    set(() => ({
      store: null,
    }));
    set(() => ({
      showBack: false,
    }));
    set(() => ({
      pageIndex: 0,
    }));
    set(() => ({
      isChildModal: false,
    }));
    modalBody = null;
  },
  showChildModal: () => {
    set(() => ({
      isChildModal: true,
    }));
  },
  dismissChildModal: () => {
    set(() => ({
      isChildModal: false,
    }));
  },
}));
let modalBody: ReactElement | null = null;
let childModalBody: ReactElement | null | undefined = null;
let pageIndexStack: number[] = [];
export interface ModalChildProps {
  changePage?: (pageIndex: number, showBackButton?: boolean) => any;
  pageIndex?: number;
}

export const ModalContext = createContext<any>(null);

interface ModalHeaderProps {
  onClose: () => any;
}

const ModalHeader = ({ onClose }: ModalHeaderProps) => {
  const { updatepageIndex, updateShowBack, pageIndex, showBack } =
    useAppModalStore();
  const onBackPressed = () => {
    pageIndexStack.pop();
    const popIndex = pageIndexStack[pageIndexStack.length - 1];
    updatepageIndex(popIndex ?? 0);
    if (pageIndex == 0) {
      updateShowBack(false);
    }
  };

  const handleKeyDown = (e: {
    preventDefault: () => void;
    keyCode: number;
  }) => {
    e.preventDefault();
    if (e.keyCode === 13) {
      onClose();
    }
  };

  return (
    <Row className="justify-between modal-header items-center relative">
      {showBack ? (
        <Row
          tabIndex={1}
          onClick={onBackPressed}
          className="items-center focus-icon"
        >
          <Image src={leftIcon} className="modal-icon" alt="back" />
          <TextBox text="Back" className="primary-color underline" />
        </Row>
      ) : (
        <div tabIndex={1} />
      )}

      <Image
        src={bcbstSilhouletteLogo}
        className="modal-icon modal-header-logo absolute m-auto left-0 right-0"
        alt="bcbst logo"
      />
      <div
        tabIndex={0}
        className="size-8 focus-icon"
        onKeyDown={handleKeyDown}
        onClick={onClose}
      >
        <Image src={closeIcon} className="size-4" alt="close" />
      </div>
    </Row>
  );
};

const ModalFooter = () => {
  return (
    <Row className="justify-center modal-footer">
      <Image
        src={bcbstSilhouletteLogo}
        className="modal-icon"
        alt="bcbst logo"
      />
    </Row>
  );
};

export const AppModal = () => {
  const {
    dismissModal,
    updateShowBack,
    updatepageIndex,
    store,
    showModal,
    pageIndex,
    isChildModal,
    showChildModal,
    dismissChildModal,
    isChildActionAppModalProp,
  } = useAppModalStore();
  const changePage = (pageIndex: number, showBackButton: boolean = false) => {
    updateShowBack(showBackButton);
    updatepageIndex(pageIndex);
    pageIndexStack.push(pageIndex);
  };
  const closeModal = () => {
    isChildActionAppModalProp ? showChildModal() : dismissModal();
  };

  const closeChildModal = () => {
    dismissChildModal();
  };
  const closeAllModal = () => {
    dismissModal();
  };
  return (
    <ModalContext.Provider value={store}>
      <InnerAppModal
        showModal={showModal}
        modalBody={modalBody ?? <></>}
        changePage={changePage}
        pageIndex={pageIndex}
        closeModal={closeModal}
        isChildModal={isChildModal}
        closeChildModal={closeChildModal}
        closeAllModal={closeAllModal}
        childModalBody={childModalBody ?? <></>}
      />
    </ModalContext.Provider>
  );
};

type InnerAppModalProps = {
  showModal?: boolean;
  closeModal?: () => any;
  modalBody: ReactElement;
  changePage?: (pageIndex: number, showBackButton: boolean) => any;
  pageIndex?: number;
  isChildModal?: boolean;
  closeChildModal?: () => any;
  closeAllModal?: () => any;
  childModalBody?: ReactElement;
};

export const InnerAppModal = ({
  showModal = true,
  closeModal = () => {},
  modalBody,
  changePage = () => {},
  pageIndex = 0,
  isChildModal = false,
  childModalBody,
}: InnerAppModalProps) => {
  return (
    <Modal
      classNames={{
        overlay: 'modal-overlay',
      }}
      showCloseIcon={false}
      open={showModal}
      closeOnOverlayClick={true}
      onClose={closeModal}
    >
      <FocusTrap>
        <div>
          <Column className="modal-container">
            <ModalHeader onClose={closeModal} />
            <Spacer size={32} />
            <div className="flex flex-row flex-grow justify-center">
              <div className="flex flex-row justify-center modal-content">
                {modalBody ? (
                  React.cloneElement(modalBody, {
                    changePage,
                    pageIndex: pageIndex,
                  })
                ) : (
                  <div />
                )}
              </div>
            </div>
            {isChildModal && (
              <div className="child-modal">
                <div className="child-modal-content">{childModalBody}</div>
              </div>
            )}
            <Spacer size={32} />
            <ModalFooter />
          </Column>
        </div>
      </FocusTrap>
    </Modal>
  );
};
