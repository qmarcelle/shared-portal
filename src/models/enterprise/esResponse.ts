import { ESDetails } from './esDetails';

export interface ESResponse<T> {
  data?: T;
  details?: ESDetails;
  errorCode?: string;
}

export interface ESResponseValidation {
  details?: ESDetails;
  errorCode?: string;
}
