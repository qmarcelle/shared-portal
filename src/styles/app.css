@tailwind utilities;

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

@font-face {
  font-family: 'Univers-45';
  src: url(./UniversLTPro-55Roman.woff2);
}

#blueback {
  background: radial-gradient(
      ellipse farthest-corner at right top,
      #5dc1fd 0%,
      #0000 60%
    ),
    linear-gradient(245deg, #067dac 0%, #00497e 70%);
  padding-bottom: 18%;
}

#logo {
  width: 175px;
  margin: 15px;
  display: inline-block;
}

#blueback #mainSection h1 {
  font-size: 24px;
  margin-top: 5px;
}

#blueback h3 {
  font-size: 20px;
  margin-top: 25px;
}

#marginSection {
  width: 416px;
  margin: 0 auto;
}

#mainSection {
  border: 1px solid #cccccc;
  border-radius: 8px;
  background-color: #ffffff;
  padding: 25px 25px;
  margin-top: 18px;
}

.buttonlink {
  color: #ffffff;
  background-color: unset;
  border: none;
  text-decoration: underline;
  margin-top: 25px;
}

#backButton::before {
  content: url('../../public/assets/icon-link_pointer_white.png');
  transform: rotate(180deg);
  display: inline-block;
  margin-right: 5px;
}

#contactbutton,
#resendbutton,
#choosebutton {
  color: #005eb9;
  font-weight: bold;
}

#confirmbutton {
  margin-top: 30px;
  width: 100%;
  height: 36px;
  background-color: #7faedc;
  border: 1px solid #7faedc;
  color: #ffffff;
  border-radius: 3px;
  opacity: 0.7;
}

#choosebutton {
  margin: 0 auto;
  margin-top: 30px;
  margin-bottom: 39px;
  width: 98%;
}

#contactbutton {
  margin: 0px;
  display: inline-block;
  padding: 0;
}

.headerbutton {
  float: right;
  margin: 21px !important;
}

@media (max-width: 767.9px) {
  #marginSection {
    width: 90%;
  }
}
.priorAuthDetail .success,
.priorAuthDetail .error,
.priorAuthDetail .partialapproval,
.priorAuthDetail .neutral {
  width: -moz-fit-content;
  width: fit-content;
}
.priorAuthDetail .statusDetail {
  margin-left: 70px;
}

.access-code {
  width: 114px;
  height: 63px;
  background-color: #e7f6ff;
}
