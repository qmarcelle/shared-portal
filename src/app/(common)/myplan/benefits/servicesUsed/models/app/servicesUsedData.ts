import { ServicesUsedItem } from '@/models/app/servicesused_details';
import { UIUser } from '@/models/app/uiUser';

export type ServicesUsedData = {
  members: UIUser[];
  services: Map<string, ServicesUsedItem[]>;
};
