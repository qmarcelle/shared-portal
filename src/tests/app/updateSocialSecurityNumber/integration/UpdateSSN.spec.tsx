import { MemberList } from '@/app/myPlan/updateSocialSecurityNumber/models/app/memberList';
import UpdateSocialSecurityNumberPage from '@/app/myPlan/updateSocialSecurityNumber/page';
import { AppModal } from '@/components/foundation/AppModal';
import { ESResponse } from '@/models/enterprise/esResponse';
import '@testing-library/jest-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { mockedAxios } from '../../../__mocks__/axios';

jest.mock('src/auth', () => ({
  auth: jest.fn(() =>
    Promise.resolve({
      user: {
        currUsr: {
          plan: { memCk: '123456789', grpId: '87898', sbsbCk: '654567656' },
        },
      },
    }),
  ),
}));

const renderUI = async () => {
  const result = await UpdateSocialSecurityNumberPage();
  // return render([result, <AppModal key={1} />]);
  return render(
    <>
      <AppModal />
      {result}
    </>,
  );
};
describe('Update SSN', () => {
  it('should render the screens correctly', async () => {
    mockedAxios.get.mockResolvedValueOnce({
      data: {
        members: [
          {
            birthDate: '06/29/2009',
            memberCk: 91722406,
            firstName: 'CHRISTMAS',
            lastName: 'HALL',
            hasSocial: true,
          },
          {
            birthDate: '06/15/2009',
            memberCk: 91722406,
            firstName: 'KRISSY',
            lastName: 'HALL',
            hasSocial: false,
          },
          {
            birthDate: '10/31/2011',
            memberCk: 91722406,
            firstName: 'CHRISTIAN',
            lastName: 'HALL',
            hasSocial: true,
          },
        ],
      } as ESResponse<MemberList>,
    });
    const component = await renderUI();
    // Init Screen is rendered correctly
    await waitFor(() => {
      expect(
        screen.getByRole('heading', {
          name: 'Update Social Security Number',
        }),
      ).toBeVisible();
      expect(screen.getByText('CHRISTMAS HALL')).toBeVisible();
    });

    mockedAxios.post.mockResolvedValueOnce({
      data: {
        message: 'Updated SSN',
      },
    });

    mockedAxios.get.mockResolvedValueOnce({
      data: {
        members: [
          {
            birthDate: '06/29/2009',
            memberCk: 91722406,
            firstName: 'CHRISTMAS',
            lastName: 'HALL',
            hasSocial: true,
          },
          {
            birthDate: '06/15/2009',
            memberCk: 91722406,
            firstName: 'KRISSY',
            lastName: 'HALL',
            hasSocial: true,
          },
          {
            birthDate: '10/31/2011',
            memberCk: 91722406,
            firstName: 'CHRISTIAN',
            lastName: 'HALL',
            hasSocial: true,
          },
        ],
      } as ESResponse<MemberList>,
    });

    const button = screen.getAllByAltText(/link/i);
    fireEvent.click(button[0]);
    await waitFor(() => {
      screen.getAllByText('Update or Add a Social Security Number');
      const text = screen.getAllByText(/Social Security Number/i);
      expect(text[0]).toBeInTheDocument();
    });
    expect(screen.queryAllByText('A SSN was found on file.')).toHaveLength(2);
    expect(screen.queryAllByText('No SSN on file.')).toHaveLength(1);
    const ssnEntryInput = screen.getByRole('textbox', {
      name: 'Social Security Number',
    });
    await userEvent.clear(ssnEntryInput);
    await userEvent.type(ssnEntryInput, '111223344');
    fireEvent.click(screen.getByRole('button', { name: /Save Changes/i }));

    await waitFor(() => {
      expect(mockedAxios.post).toHaveBeenCalledWith(
        '/api/member/v1/members/byMemberCk/123456789/updateSSN',
        {
          ssn: '111223344',
        },
      );
    });
    expect(
      screen.getByText(
        'Your social security number has been successfully updated.',
      ),
    ).toBeVisible();

    fireEvent.click(screen.getByRole('button', { name: /Done/i }));

    await waitFor(() => {
      expect(mockedAxios.get).toHaveBeenCalledTimes(2);
      expect(
        screen.queryAllByText('A SSN was found on file.')[0],
      ).toBeVisible();
      expect(screen.queryAllByText('A SSN was found on file.')).toHaveLength(3);
    });

    expect(component.baseElement).toMatchSnapshot();
  });
});
