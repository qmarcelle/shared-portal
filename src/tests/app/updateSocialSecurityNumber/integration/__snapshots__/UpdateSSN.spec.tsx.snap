// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update SSN should render the screens correctly 1`] = `
<body
  style=""
>
  <div>
    <div
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col app-content app-base-font-color"
      >
        <h1
          class="title-1 font-bold"
        >
          Update Social Security Number
        </h1>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
        />
        <p
          class="body-1"
          style="display: block; max-width: 100%;"
        >
          You can update the Social Security Number (SSN) we have on file here.
        </p>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main small-section"
            >
              <div
                class="flex flex-col"
              >
                <h1
                  class="title-1 mb-0 title-2"
                >
                  About Social Security Number
                </h1>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <p
                  class="body-1 body-1 mb-0"
                  style="display: block; max-width: 100%;"
                >
                  For privacy and security, we will not display the current social security number we have on file.
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <p
                  class="body-1 body-1 mb-0"
                  style="display: block; max-width: 100%;"
                >
                  Your SSN is used to process important tax related documents. Adding or updating your SSN will ensure that your documents are accurate and timely.
                </p>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="flex flex-col flex flex-col mt-4"
            >
              <div
                class="flex flex-col flex flex-col"
              >
                <div
                  class="card-elevated cursor-pointer mb-4"
                >
                  <div
                    class="flex flex-col m-4"
                  >
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 ml-2 font-bold title-3"
                      style="display: block; max-width: 100%;"
                    >
                      CHRISTMAS HALL
                    </p>
                    <p
                      class="body-1 ml-2"
                      style="display: block; max-width: 100%;"
                    >
                      DOB: 06/29/2009
                    </p>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class=""
                      style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                    />
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 ml-2"
                      style="display: block; max-width: 100%;"
                    >
                      A SSN was found on file.
                    </p>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class="flex flex-row"
                    >
                      <p
                        class="font-bold primary-color ml-2"
                      >
                        Update
                      </p>
                      <div
                        class="ml-2"
                      >
                        <img
                          alt="link"
                          data-nimg="1"
                          decoding="async"
                          height="40"
                          loading="lazy"
                          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                          style="color: transparent;"
                          width="40"
                        />
                      </div>
                    </div>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                  </div>
                </div>
                <div
                  class="card-elevated cursor-pointer mb-4"
                >
                  <div
                    class="flex flex-col m-4"
                  >
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 ml-2 font-bold title-3"
                      style="display: block; max-width: 100%;"
                    >
                      KRISSY HALL
                    </p>
                    <p
                      class="body-1 ml-2"
                      style="display: block; max-width: 100%;"
                    >
                      DOB: 06/15/2009
                    </p>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class=""
                      style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                    />
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 ml-2"
                      style="display: block; max-width: 100%;"
                    >
                      A SSN was found on file.
                    </p>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class="flex flex-row"
                    >
                      <p
                        class="font-bold primary-color ml-2"
                      >
                        Update
                      </p>
                      <div
                        class="ml-2"
                      >
                        <img
                          alt="link"
                          data-nimg="1"
                          decoding="async"
                          height="40"
                          loading="lazy"
                          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                          style="color: transparent;"
                          width="40"
                        />
                      </div>
                    </div>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                  </div>
                </div>
                <div
                  class="card-elevated cursor-pointer mb-4"
                >
                  <div
                    class="flex flex-col m-4"
                  >
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 ml-2 font-bold title-3"
                      style="display: block; max-width: 100%;"
                    >
                      CHRISTIAN HALL
                    </p>
                    <p
                      class="body-1 ml-2"
                      style="display: block; max-width: 100%;"
                    >
                      DOB: 10/31/2011
                    </p>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class=""
                      style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                    />
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <p
                      class="body-1 ml-2"
                      style="display: block; max-width: 100%;"
                    >
                      A SSN was found on file.
                    </p>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                    <div
                      class="flex flex-row"
                    >
                      <p
                        class="font-bold primary-color ml-2"
                      >
                        Update
                      </p>
                      <div
                        class="ml-2"
                      >
                        <img
                          alt="link"
                          data-nimg="1"
                          decoding="async"
                          height="40"
                          loading="lazy"
                          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                          style="color: transparent;"
                          width="40"
                        />
                      </div>
                    </div>
                    <span
                      style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
  <div>
    <div
      class="react-responsive-modal-root"
      data-testid="root"
    >
      <div
        aria-hidden="true"
        class="react-responsive-modal-overlay modal-overlay"
        data-testid="overlay"
        style="animation: react-responsive-modal-overlay-out 300ms;"
      />
      <div
        class="react-responsive-modal-container react-responsive-modal-containerCenter"
        data-testid="modal-container"
      >
        <div
          aria-modal="true"
          class="react-responsive-modal-modal"
          data-testid="modal"
          role="dialog"
          style="animation: react-responsive-modal-modal-out 300ms;"
          tabindex="-1"
        >
          <div>
            <div
              class="flex flex-col modal-container"
            >
              <div
                class="flex flex-row justify-between modal-header items-center relative"
              >
                <div
                  tabindex="1"
                />
                <img
                  alt="bcbst logo"
                  class="modal-icon modal-header-logo absolute m-auto left-0 right-0"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
                <div
                  class="size-8 focus-icon"
                  tabindex="0"
                >
                  <img
                    alt="close"
                    class="size-4"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row flex-grow justify-center"
              >
                <div
                  class="flex flex-row justify-center modal-content"
                >
                  <div />
                </div>
              </div>
              <span
                style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
              />
              <div
                class="flex flex-row justify-center modal-footer"
              >
                <img
                  alt="bcbst logo"
                  class="modal-icon"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
